src/anyio/_core/_fileio.py

```diff
@@ -73,16 +156,16 @@ async def __aiter__(self) -> AsyncIterator[bytes]:
     async def aclose(self) -> None:
         return await to_thread.run_sync(self._fp.close)
 
-    async def read(self, size: int = -1) -> Union[bytes, str]:
+    async def read(self, size: int = -1) -> T_IO:
         return await to_thread.run_sync(self._fp.read, size)
 
-    async def read1(self, size: int = -1) -> Union[bytes, str]:
+    async def read1(self, size: int = -1) -> bytes:
         return await to_thread.run_sync(self._fp.read1, size)
 
-    async def readline(self) -> bytes:
+    async def readline(self) -> T_IO:
         return await to_thread.run_sync(self._fp.readline)
 
-    async def readlines(self) -> bytes:
+    async def readlines(self) -> T_IO:
```

Comment:
```suggestion
    async def readlines(self) -> Iterable[T_IO]:
```

Reply:
I'll go with `List[AnyStr]` instead since typeshed defines `readlines()` to return `List`.